src/Simplex_tree/include/gudhi/Simplex_tree/serialization_utils.h

```diff
@@ -0,0 +1,58 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Vincent Rouvreau
+ *
+ *    Copyright (C) 2023 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+#ifndef SIMPLEX_TREE_SERIALIZATION_UTILS_H_
+#define SIMPLEX_TREE_SERIALIZATION_UTILS_H_
+
+#include <cstring>  // for memcpy and std::size_t
+#include <iostream>
+#include <iomanip>
+#include <memory>
+
+namespace Gudhi {
+
+namespace simplex_tree {
+
+/** \addtogroup simplex_tree 
+ * Serialization and deserialization utils for the Simplex_tree.
+ *  @{
+ */
+
+/** \brief Serialize the given value and insert it at the end of the buffer.
+ */
+template<class ArgumentType>
+void serialize(ArgumentType value, std::vector<char>& buffer) {
+  const std::size_t arg_size = sizeof(ArgumentType);
+  char serial[arg_size];
+  memcpy(&serial, &value, arg_size);
+  buffer.insert(std::end(buffer), std::begin(serial), std::end(serial));
+}
+
+/** \brief Deserialize at the start position in an array of char and sets the value with it.
+ * 
+ * @return The new position in the array of char.
+ * 
+ * @warning It is the user resposibility to ensure that the array of char is wide enough.
+ */
+template<class ArgumentType>
+std::vector<char>::const_iterator deserialize(std::vector<char>::const_iterator start, ArgumentType& value) {
```

Comment:
Done with the change of interface (use `num_simplices()` instead of `vector::insert`) on [04e5a85](https://github.com/GUDHI/gudhi-devel/pull/810/commits/04e5a8541cc11e79d5c69aa7e0d7ad965c9a88d2)